text/0000-default-field-values.md

```diff
@@ -0,0 +1,1933 @@
+```
+
+This could be interpreted as `open_window({ RangeFull })`, see the previous
+section for a discussion... alternatively `open_window(_)` could be permitted
+instead for general value inference where `_` is a placeholder expression
+similar to `_` as a type expression placeholder
+(i.e. a fresh and unconstrained unification variable).
+
+If you wanted to override a default, you would write:
+
+```rust
+open_window({ height: 720, });
+```
+
+Note that the syntax used to give fields in structural records defaults belongs
+to the type grammar; in other words, the following would be legal:
+
+```rust
+type RGB = { red: u8 = 0, green: u8 = 0, blue: u8 = 0 };
+
+let color: RGB = { red: 255, };
+```
+
+As structural records are not yet in the language,
+figuring out designs for how to extend this RFC to them is left
+as possible work for the future.
+
+
+## Integration with struct literal type inference
+
+Yet another common requested feature is the introduction of struct literal type
+inference in the form of elision of the name of an ADT literal when it can be
+gleaned from context. This has sometimes been proposed as an alternative or
+complementary to structural records. This would allow people to write
+`foo(_ { bar: 42 })` where the function argument type is inferred from the `foo`
+definition. struct literal type inference with default struct fields would also
+allow people to write APIs that "feel" like named function arguments when
+calling them, although not when defining them.
+
+```rust
+struct Config {
+    height: u32 = 1080,
+    width: u32 = 1920,
+}
+fn open_window(config: Config) {
+    // logic...
+}
+
+open_window(_ { width: 800, .. });
+```
+
+## Accessing default values from the type
+
+If one were to conceptualize default field values in the following way:
+
+```rust
+struct Config {
+    height: u32 = Self::HEIGHT,
+    width: u32 = Self::WIDTH,
+}
+
+impl Config {
+    const HEIGHT: u32 = 1080,
+    const WIDTH: u32 = 1920,
+}
+```
+
+It would follow that one should be able to access the value of these defaults
+*without* constructing `Config`, by writing `Config::HEIGHT`. I do not believe
+this should be done or advanced, but there's nothing in this RFC that precludes
+*some* mechanism to access these values in the future. With the RFC as written,
+these values can be accessed by instantiating `Config { .. }.height`, as long
+as `height` is visible in the current scope.
+
+Note that the *opposite* is supported, writing that code will compile, so any
+API author that wants to make these `const` values on the type can (only
+restriction is that `Self` isn't accepted in current `const` contexts):
+
+```rust
+struct Config {
+    height: u32 = Config::HEIGHT,
+    width: u32 = Config::WIDTH,
+}
+
+impl Config {
+    const HEIGHT: u32 = 1080,
+    const WIDTH: u32 = 1920,
+}
+```
+## Non-const values
+
+[strong reasons]: #on-const-contexts
+
+Although there are [strong reasons] to restrict default values only to const
```

Comment:
One strong class of non-const values is *other* default implementations. Without const trait implementations, this will be an annoying restriction.

Reply:
Added note of this.